assets/js/main.js

```diff
@@ -0,0 +1,412 @@
+    button.textContent = content
+    button.addEventListener('click', function () {
+      callback(appendTo)
+    })
+    appendTo.appendChild(button)
+  }
+
+  function toMainMenu (windowToClose) {
+    close(windowToClose)
+    mainMenu.style.display = ''
+    innerContainer.style.border = ''
+  }
+
+  function restart (windowToClose) {
+    close(windowToClose)
+    gameSetup()
+  }
+
+  function close (windowToClose) {
+    container.removeChild(windowToClose)
+  }
+
+  function setLife () {
+    if (lives === 0) {
+      innerContainer.innerHTML = ''
+      var audio = document.createElement('audio')
+      audio.src = 'assets/audio/gameover.wav'
+      audio.autoplay = true
+      container.appendChild(audio)
+      inGameMessage('Game Over!', 1000)
+      setTimeout(createGameoverWindow, 2000)
+    }
+  }
+
+  function addHeart (appendTo) {
+    var heart = document.createElement('img')
+    heart.src = 'assets/images/redheart.png'
+    heart.setAttribute('class', 'heart')
+    appendTo.appendChild(heart)
+  }
+
+  function setScore () {
+    var scoreCounter = document.querySelector('#scoreCounter')
+    scoreCounter.textContent = 'Score: ' + score
+    var scoreToProgress = document.querySelector('#scoreToProgress')
+    var pointsToProgress = 5000 * level - score
+    scoreToProgress.textContent = pointsToProgress + ' points to next level'
+    if (score >= 5000 * level) {
+      if (combo % 50 !== 0) inGameMessage('Level up!', 500)
+      else {
+        setTimeout(inGameMessage, 1010, 'Level up!', 500)
+      }
+      setLevel()
+    }
+  }
+
+  function setCombo () {
+    if (combo > maxCombo) maxCombo = combo
+    if (combo >= 10) {
+      var comboCounter = document.querySelector('#comboCounter')
+      comboCounter.textContent = 'Combo ' + combo
+      comboCounter.style.top = '50px'
+      comboCounter.style.transition = ''
+      comboCounter.style.opacity = '1'
+      setTimeout(function () {
+        comboCounter.style.opacity = '0'
+        comboCounter.style.transition = 'opacity 1s, top 0.1s'
+        comboCounter.style.top = '45px'
+      }, 10)
+      if (combo % 50 === 0) {
+        score += 1000
+        inGameMessage('Nice!', 10)
+        setScore()
+      }
+    }
+  }
+
+  function createInGameMessage () {
+    var inGameMessage = document.createElement('div')
+    inGameMessage.setAttribute('id', 'inGameMessage')
+    container.appendChild(inGameMessage)
+  }
+
+  function inGameMessage (message, durationInMs) {
+    var inGameMessage = document.querySelector('#inGameMessage')
+    inGameMessage.textContent = message
+    inGameMessage.style.opacity = '1'
+    inGameMessage.style.transition = ''
+    setTimeout(function () {
+      inGameMessage.style.opacity = '0'
+      inGameMessage.style.transition = 'opacity 1s'
+    }, durationInMs)
+  }
+
+  function randomNo (min, max) {
+    min = Math.ceil(min)
+    max = Math.floor(max)
+    return Math.floor(Math.random() * (max - min + 1)) + min
```

Comment:
min max is because randomNo is expecting a minimum number and maximum number to be passed so that it can return a random number between 'min' and 'max'.

There's one specific case where I need to round up or down to the nearest integer: When I determine the length of the word to type based on what level it is (line 279). For  example  randomNoRounded(3 + level / 3, 4 + level / 3) on line 289 will be (3 1/3 , 4 1/3) on level 1, ( 3 2/3, 4 2/3) on level 2 so on and so forth.  So if i use math round to round the returned number to the nearest variable, on level 1 there is like around 17% chance of 3 letter word and 83% chance of 4 letter word. level 2 there is 83% chance of 4 letter word and 17% chance of 5 letter word. But the way my randomNo  is defined, if i do randomNo(3 1/3, 4 1/3) it will return 4 100% of the time, which is not the desired result.